os/kernel/irq/fin_notify.c

```diff
@@ -0,0 +1,77 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdbool.h>
+#include <arch/irq.h>
+#include <sys/types.h>
+#include <tinyara/arch.h>
+#include <tinyara/irq.h>
+#include <tinyara/sched.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: fin_notify
+ *
+ * Description:
+ *   This function sends an irq event to specific pid task.
+ *   If that task was in g_waitingforfin list, it will be unblocked when receiving irq event.
```

Comment:
What if the task is waiting for other resources such as semaphore or message?

Reply:
In that case, it will not be unblocked.
I will add the explanation about this.
Thanks.